### PR TITLE
Payments refactor

### PR DESCRIPTION
**Pull request by mdomasevicius**

### CODE REVIEW

File: src/Components/Payments/gateways/PayPal.tsx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        const rates = await Promise.all(
+          currencies.map((c) => api.exchangeRate(c).catch(() => ({ currency: c, amount: 0 } as Money))),
+        )
+
+        setState((p) => ({
+          ...p,
+          taxCountry,
+          order: orders.find((o) => o.status === 'new' && o.gatewayName === gateway.name),
+          fiatToMystEther: _.chain(
+            rates.map((r) => ({ currency: r.currency, mystEther: myst.toBig(0.77).div(r.amount).toFixed(2) })),
```
Comment (Guillembonet) [inline src/Components/Payments/gateways/PayPal.tsx:77]: not sure what this 0.77 refers to here? 0.77 MYST? 0.77 whatever currency we use?

File: src/Pages/Onboarding/steps/NetworkRegistration.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+    skipIfPaid()
+  }, [])
+
+  useEffect(() => {
+    const interval = setInterval(() => {
+      api.identityBalanceRefresh(identity.id)
+      api.registrationPayment(identity.id).then((r) => setRegistrationPayment(r))
+    }, 5 * 1000)
+    return () => clearInterval(interval)
```
Comment (Guillembonet) [inline src/Pages/Onboarding/steps/NetworkRegistration.tsx:78]: won't this stop the interval?

File: src/commons/page.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2022 BlockDev AG
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const refreshPage = async (seconds: number = 0) => {
+  if (seconds > 0) {
+    await new Promise((r) => setInterval(r, seconds * 1_000))
```
Comment (Guillembonet) [inline src/commons/page.ts:10]: not sure I understand the purpose of this, shouldn't it be setTimeout in this case?

**Review by Guillembonet — APPROVED** [review]
some minor comments, if its tested and works fine I don't see a problem with it

File: src/Pages/Onboarding/steps/NetworkRegistration.tsx (hunk 2 above)
Comment (mdomasevicius) [reply src/Pages/Onboarding/steps/NetworkRegistration.tsx:78]: if the component get's unmounted

File: src/Components/Payments/gateways/PayPal.tsx (hunk 1 above)
Comment (mdomasevicius) [reply src/Components/Payments/gateways/PayPal.tsx:77]: I will export 0.77 to constant with meaningful name, but the logic is that we want to ask user to pay around 1 eur/usd/gbp
0.77 is what we ask to pay (+ paypal adds VAT) so this is what it is: 0.77 * rate.amount

image

so if 1 myst is 0.287077 USD -> 0.77 / 0.287077 = 2.682207213 (MYST Ethers), this value is sent to /create-order endpoint

Comment (Guillembonet) [reply src/Components/Payments/gateways/PayPal.tsx:77]: I think it will be fine

File: src/commons/page.ts (hunk 3 above)
Comment (mdomasevicius) [reply src/commons/page.ts:10]: forgot that `setTimeout` existed - will revise in next iteration of changes

Comment (Guillembonet) [reply src/commons/page.ts:10]: maybe also check that all `setInterval` are okay and cannot be `setTimeout` instead

Comment (mdomasevicius) [reply src/commons/page.ts:10]: I will revise all such parts with next iteration of refactoring

**Pull request merged**